pkg/controller/knativeserving/common/extensions.go

```diff
@@ -56,8 +57,17 @@ func (exts Extensions) Transform(instance *servingv1alpha1.KnativeServing) []mf.
 	for _, extension := range exts {
 		result = append(result, extension.Transformers...)
 	}
-	// Let any config in instance override everything else
 	return append(result, func(u *unstructured.Unstructured) error {
+		// Update the deployment with the new registry and tag
+		if u.GetKind() == "Deployment" {
+			registry := instance.Spec.Image.Image
+			tag := instance.Spec.Image.Tag
+			var deployment *appsv1.Deployment
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, deployment); err == nil {
```

Comment:
I've been using `scheme.Convert` to do this in other places, but I'm not sure if either is preferred. It makes me think maybe the controller's scheme should be made available in this function though.

Reply:
Done